### DISCUSSION TITLE
How Do I Retrieve All TAFs For Current Time?

### DISCUSSION THREAD

**Discussion by danbo0001** [weather-gov/api#567] [Q&A] [answered]
This is my first look at these api's so I am trying to work my way through how to use them :)

I can get a list of TAFs using:-

https://api.weather.gov/products/types/taf

but this seems to return multiple TAFs per station per issuance time, as well as older TAFs from earlier in the day and the day before (I am assuming here that newer TAFs for a station override older ones).

Also, the records returned don't seem to include the actual TAF itself, instead they have a link with a guid that can be used to get the text itself - so if there are 1000 TAFs, then it looks like I need to do one call to get the list then 1000 calls to get the individual TAFs. Is there any way to get them all in one call?

**Comment by dwhitemv25** [comment, marked as answer]
You can download TAFs in bulk from the aviationweather.gov data server:

https://www.aviationweather.gov/dataserver

Be sure to read the 'Best Practices' page about making large volume requests.

**Reply by danbo0001** [reply]
that's perfect! many thanks :)
